### PR TITLE
Add validate_slength's optional 3rd arg to README

### PR DESCRIPTION
**Pull request by alexjfisher**
I was thinking about creating a validate_non_empty_string function or a validate_minimum_slength function when I discovered validate_slength could already take a minimum string length as its 3rd argument.

### CODE REVIEW

File: lib/puppet/parser/functions/validate_slength.rb [hunk 1]
```diff
@@ -44,7 +44,7 @@ module Puppet::Parser::Functions
     end
 
     if min_length > max_length
-      raise Puppet::ParseError, "validate_slength(): Expected second argument to be larger than third argument"
+      raise Puppet::ParseError, "validate_slength(): Expected second argument to be equal to or larger than third argument"
```
Comment (DavidS) [inline lib/puppet/parser/functions/validate_slength.rb]: That does not match the `>` in L46

Comment (alexjfisher) [inline lib/puppet/parser/functions/validate_slength.rb]: The second argument is 'max_length' and the third is 'min_length'.  Or are you suggesting reversing the comparison in L46 for clarity?

Comment (DavidS) [inline lib/puppet/parser/functions/validate_slength.rb]: Yeah, that threw me off. Text and code should always align.

**Comment by alexjfisher** [comment]
@DavidS I've added a commit that hopefully improves the clarity of the code.

There is already a spec test for the corner case of min_length == max_length.
https://github.com/puppetlabs/puppetlabs-stdlib/blob/master/spec/functions/validate_slength_spec.rb#L54

**Comment by alexjfisher** [comment]
BTW, what do you think about adding a validate_non_empty_string function?
It would be a lot neater that validate_slength("string",999999,1)

**Comment by DavidS** [comment]
closing as merged in #482
